doc/lsst.pipe.base/command-line-task-parallel-howto.rst

```diff
@@ -0,0 +1,97 @@
+=============================================
+
+Command-line tasks can process multiple data IDs in parallel across multiple CPUs of one machine if you provide a :option:`-j`/:option:`--processes` argument with a value greater than ``1``.
+For example, to spread processing across up to eight CPUs:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --id -j 8 ...
+
+When operating with :option:`-j`, command-line tasks have a default per-process timeout of 720 hours.
+You can change this timeout via the :option:`--timeout` command-line argument.
+
+Log messages from each process are combined into the same console output stream.
+To discern logging statements from each process use the :option:`--longlog` argument to enable a verbose logging format that includes the data ID being processed.
+See :ref:`command-line-task-logging-howto-longlog` for more information.
+
+.. note::
+
+   Internally, :option:`-j`-type multiprocessing is implemented with the `multiprocessing` module in the Python standard library.
+
+   If a task does not support :option:`-j` parallel processing it will report a non-fatal warning:
+
+   .. code-block:: text
+
+      This task does not support multiprocessing; using one process
+
+.. _command-line-task-parallel-howto-distributed:
+
+How to enable distributed processing
+====================================
+
+Command-line tasks implemented in the ``pipe_drivers`` package can distribute processing across multiple nodes using MPI or batch submission to clusters (PBS or SLURM).
+See the ``pipe_drivers`` documentation for more information.
+
+.. _command-line-task-parallel-howto-threading:
+
+How to control threading while parallel processing
+==================================================
+
+LSST Science Pipelines uses low-level math libraries that support threading.
+OpenBLAS_ and MKL_ default to using the same number of threads as CPUs present on the machine.
+When parallel processing at a higher level (using :option:`-j`, for example), threading in these libraries can result in a **net decrease in performance** because of thread contention.
+
+When running with :option:`-j` multiprocessing or via ``ctrl_pool``, command-line tasks check if low-level libraries are using their default threading behavior (same number of threads as CPUs).
+If so, the task will automatically disable multi-threading in these libraries.
+The command-line task also sends a warning when it does so:
+
+.. code-block:: text
+
+   WARNING: You are using OpenBLAS with multiple threads (16), but have not specified the number of threads using one of the OpenBLAS environment variables: OPENBLAS_NUM_THREADS, GOTO_NUM_THREADS, OMP_NUM_THREADS. This may indicate that you are unintentionally using multiple threads, which may cause problems. WE HAVE THEREFORE DISABLED OpenBLAS THREADING. If you know what you are doing and want threads enabled implicitly, set the environment variable LSST_ALLOW_IMPLICIT_THREADS.
+
```

Comment:
It might be worth mentioning that these considerations are independent of `pipe_drivers` use of MPI; we're talking multi-threading here (openMP)

Reply:
@RobertLuptonTheGood I'm not following. Are you saying the intoducing paragraph is wrong:

> When running with :option:`-j` multiprocessing or via ``ctrl_pool``, command-line tasks check if low-level libraries are using their default threading behavior (same number of threads as CPUs).
If so, the task will automatically disable multi-threading in these libraries.
The command-line task also sends a warning when it does so: